### ISSUE TITLE
Include DOSBox-VGM patch

### ISSUE THREAD

**Issue by Burrito78** [dosbox-staging/dosbox-staging#2973]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

None

### Is your feature request related to a problem? Please describe.

Possibly include NewRisingSun's DOSBox-VGM patch in Staging.
https://vgmrips.net/forum/viewtopic.php?t=2585

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by MasterO2** [comment]
The VGM Rips Forum now knows about Staging as well:

https://vgmrips.net/forum/viewtopic.php?p=17039#p17039

That should bring a few people over here in the future.

**Comment by Burrito78** [comment]
Related DOSBox-X request with more info:

https://github.com/joncampbell123/dosbox-x/issues/4504

**Comment by johnnovak** [comment]
Thanks for the context @MasterO2 and @Burrito78, this seems like a useful future enhancement 👍🏻

**Comment by kcgen** [comment]
The VGM format has a lot more players and plugins available on various devices (more so than `.DRO`), so I guess it makes sense for us to move to this capture format.

NewRisingSun's patch has been added as a branch off SVN's r4000 commit under **`origin/svn/r4000-nrs-vgm-patch-1`** (of course, it still needs to be carried forward as a PR to get in Staging `main`).

NewRisingSun's patch also includes critical fixes to the GameBlaster and Tandy emulation code, surely to prevent people from capturing and submitting slightly broken VGM recordings:

> _"The source code diff also incorporates my [correction to the Game Blaster base clock](http://www.vogons.org/viewtopic.php?f=41&t=38350), [my patch adding support for Tandy 1000 direct DAC writes](http://www.vogons.org/viewtopic.php?f=41&t=51382), and [my patch correcting the interpretation of SN76496 frequency register value 0](http://www.vogons.org/viewtopic.php?f=41&t=37917)."_

Of these three, I believe only the Tandy direct DAC writes is still needed.

The GameBlaster's clock and operating frequency are OK in latest SVN and Staging (slightly different fixes in both repos at different times, but corrected none the less).

Regarding the Tandy PSG SN76496 frequency register value 0 fix, at the time  DOSBox used its own PSG implementation.

- Here's the original SVN r4000 (Oct 2016) in Zak McKracken's intro:

    https://github.com/dosbox-staging/dosbox-staging/assets/1557255/53e9623b-7d5d-4719-8b8c-fec34d5e7bec

- Here's r4000 plus NRS's low-frequency fix (Jan 2017):

    https://github.com/dosbox-staging/dosbox-staging/assets/1557255/0bfce16b-f607-4bc6-8f57-b7458f05db54

- Here's [Nerdly Pleasures](https://www.youtube.com/@NerdlyPleasures)'s recording from a real Tandy PC, with comments:

    > _"It should be noted that the Tandy 1000s subject the audio output to a 4.8 KHz first order lowpass filter.  ScummVM"s emulation of Zak's music is not quite right in its presentation of the opening's bassline, it does not take into account that a square wave period value of 0 actually means a period value of 1024."_

    > _"The Enhanced Versions run a bit slow on the 8088 and 8086, so here is a sample of them running at their ideal speed with their ideal graphics and sound.  The intro for Zak McKracken is particularly timed so that the music ends when Zak walks off the final screen with the giant face mask."_

    https://www.youtube.com/watch?v=wt-_ashF6EA&t=251s

- Eventually DOSBox SVN was migrated to MAME's SN76496 implementation, which does correctly generate a low-frequency (period value of `1024` or 0x400) on a register value of 0:

    https://github.com/dosbox-staging/dosbox-staging/assets/1557255/2552afee-38f8-4b2d-a9f0-9904beb8e5fb

- DOSBox Staging hasn't changed from SVN in this regard, so is also OK. Here's `main` with a 4.8 KHz first order lowpass filter as suggested by Nerdly Pleasures combined with `reverb` and `chorus` on:

    https://github.com/dosbox-staging/dosbox-staging/assets/1557255/9865208f-be4a-47b4-93d3-bc5b5c89acb5

    For reference, here's the conf:

    ```ini
    [render]
    cga_colors = tandy 50
    
    [dosbox]
    machine=tandy
    memsize=1
    startup_verbosity = quiet
    
    [cpu]
    cycles=920
    
    [mixer]
    reverb = on
    chorus = on
    
    [midi]
    mididevice =  none
    
    [sblaster]
    sbtype = none
    oplmode = none
    
    [ethernet]
    ne2000 = off
    
    [speaker]
    pcspeaker = off
    tandy = on
    lpt_dac = none
    ps1audio = false
    tandy_filter = lpf 1 4800
    
    [joystick]
    joysticktype = disabled
    
    [dos]
    xms = false
    ems = false
    umb = false
    
    [composite]
    composite  = off
    
    [autoexec]
    @echo off
    imgmount a floppies/disk1.img floppies/disk2.img floppies/disk3.img -t     floppy
    a:
    zak
    ```

**Comment by johnnovak** [comment]
Nice extra info @kcgen 🚀 I'm unsure whether we should keep the DRO capture option, though. I've never used it myself, but maybe keep it as a legacy option? Might be useful for some people using older tools? I have no idea, but if the new format is superior in every way, yeah maybe time to drop it.

**Comment by kcgen** [comment]
> but maybe keep it as a legacy option?

Yup - looking at the code, DRO could be retained without any issues.

**Comment by Burrito78** [comment]
Adding the files here in case the forum goes down:
[DOSBox-VGM build.zip](https://github.com/user-attachments/files/15821796/DOSBox-VGM.build.zip)
[r4000-vgmcapture.zip](https://github.com/user-attachments/files/15821797/r4000-vgmcapture.zip)
